src/galois/_polys/_primitive.py

```diff
@@ -86,35 +116,59 @@ def primitive_poly(order: int, degree: int, method: Literal["min", "max", "rando
     """
     verify_isinstance(order, int)
     verify_isinstance(degree, int)
+    verify_isinstance(terms, (int, str), optional=True)
 
     if not is_prime_power(order):
         raise ValueError(f"Argument 'order' must be a prime power, not {order}.")
     if not degree >= 1:
         raise ValueError(
             f"Argument 'degree' must be at least 1, not {degree}. There are no primitive polynomials with degree 0."
         )
+    if isinstance(terms, int) and not 1 <= terms <= degree + 1:
+        raise ValueError(f"Argument 'terms' must be at least 1 and at most {degree + 1}, not {terms}.")
+    if isinstance(terms, str) and not terms in ["min"]:
+        raise ValueError(f"Argument 'terms' must be 'min', not {terms!r}.")
     if not method in ["min", "max", "random"]:
         raise ValueError(f"Argument 'method' must be in ['min', 'max', 'random'], not {method!r}.")
 
-    if method == "min":
-        poly = next(primitive_polys(order, degree))
-    elif method == "max":
-        poly = next(primitive_polys(order, degree, reverse=True))
-    else:
-        poly = _random_search(order, degree)
+    try:
+        if method == "min":
+            poly = next(primitive_polys(order, degree, terms))
+        elif method == "max":
+            poly = next(primitive_polys(order, degree, terms, reverse=True))
+        else:
+            if terms == "min":
+                terms = _minimum_terms(order, degree, "is_primitive")
+            poly = _random_search(order, degree, terms)
```

Comment:
Maybe a future improvement for performance is not to use a random search when `terms="min"`? Or is this the best known algorithm?

Reply:
Good point.

I struggled a lot getting an efficient iteration through only the polynomials with `t` non-zero terms. I used `combinations` and `permutations` from `itertools`, but was never able to get the iteration order in precisely lexicographical order. So, instead, I created this recursive algorithm to iterate in lexicographical order.

But now that you mention this, I could resurrect my `combinations` and `permutations` code, add an extra shuffle, and iterate through that for the `method="random"` and fixed-term case. I'll work on that.

Reply:
Probably it's also a good idea to check how people using NTL find these polynomials?

Reply:
Yes, that could be helpful.